src/k2/module/k23si/Module.cpp

```diff
@@ -756,7 +751,7 @@ bool K23SIPartitionModule::_makeFieldsForDiffVersion(dto::Schema& schema, dto::S
     return true;
 }
 
-bool K23SIPartitionModule::_parsePartialRecord(dto::K23SIPartialUpdateRequest& request, std::deque<dto::DataRecord>& versions) {
+bool K23SIPartitionModule::_parsePartialRecord(dto::K23SIWriteRequest& request, std::deque<dto::DataRecord>& versions) {
```

Comment:
I don't think we need to parse the entire deque here. Just a ref to the previous version should suffice